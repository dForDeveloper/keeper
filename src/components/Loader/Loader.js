import React from 'react';

const Loader = () => {

  return (
    <div className="Loader">
      <div className="Loader--div Loader--balls"></div>
    </div>
  )
};

export default Loader;