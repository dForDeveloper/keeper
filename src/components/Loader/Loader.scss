.Loader {
  z-index: 2;
  position: absolute;
  background-color: #fff;
  width: 100%;
  height: 100vh;
  padding-top: 35vh;
}

.Loader--div {
	width: 64px;
	height: 64px;
	clear: both;
  margin: 20px auto;
  position: fixed;
  z-index: 2;
}

.Loader--balls, .Loader--balls:before, .Loader--balls:after {
	border-radius: 50%;
	background-color: #000;
	width: 18px;
	height: 18px;
	transform-origin: center center;
	display: inline-block;
}

.Loader--balls {
	position: relative;
	background-color: rgba(#000,1);
	opacity: 1;
	animation: middle 1s infinite linear;
}

.Loader--balls:before, .Loader--balls:after {
	content: '';
	position: relative;
	opacity: 0.25;
}

.Loader--balls:before {
	left: 30px;
	top: 0px;
	animation: left 1s infinite linear;
}

.Loader--balls:after {
	left: -30px;
	top: -23px;
	animation: right 1s infinite linear;
}

@keyframes middle {
	0% { opacity: 1; }
	33% {  opacity: 0.25; }
	66% { opacity: 0.25; }
	100% {  opacity: 1; }
}

@keyframes left {
	0% { opacity: 0.25; }
	33% { opacity: 1; }
	66% { opacity: 0.25; }
}

@keyframes right {
	33% { opacity: 0.25; }
	66% { opacity: 1; }
	100% { opacity: 0.25; }
}