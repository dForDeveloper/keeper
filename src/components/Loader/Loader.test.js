import React from 'react';
import { shallow } from 'enzyme';
import Loader from './Loader';

describe('Loader', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Loader />
    );
  });

  it('should pass snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  });
});