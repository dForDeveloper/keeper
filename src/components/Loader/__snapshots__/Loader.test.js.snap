// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should pass snapshot test 1`] = `
<div
  className="Loader"
>
  <div
    className="Loader--div Loader--balls"
  />
</div>
`;
